### PR TITLE
add WorkerMgt

### PR DESCRIPTION
**Pull request by xudean**

### CODE REVIEW

File: contracts/PADORegistryCoordinator.sol [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            _blsApkRegistry,
+            _indexRegistry
+        )
+    {}
+
+    /**
+     * @notice Sets the workerMgt contract address.
+     * @param _workerMgt The address of the workerMgt contract.
+     */
+    function setWorkerMgt(IWorkerMgt _workerMgt) external onlyOwner {
```
Comment (fksyuan) [inline contracts/PADORegistryCoordinator.sol:78]: The workerMgt data field can be set in initialize function.

File: contracts/PADORegistryCoordinator.sol [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     * @param socket The socket address of the worker.
+     * @param params The parameters for registering the worker's BLS public key.
+     * @param operatorSignature The signature of the operator.
+     */
+    function registerOperator(
+        bytes calldata quorumNumbers,
+        string calldata socket,
+        IBLSApkRegistry.PubkeyRegistrationParams calldata params,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) public override onlyWorkerMgt{
```
Comment (fksyuan) [inline contracts/PADORegistryCoordinator.sol]: The function must be add onlyWhenNotPaused modify.

File: contracts/WorkerMgt.sol [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+import {BN254} from "@eigenlayer-middleware/src/libraries/BN254.sol";
+
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {PADORegistryCoordinator} from "./PADORegistryCoordinator.sol";
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract WorkerMgt is IWorkerMgt, OwnableUpgradeable {
+    event WorkerRegistry(
```
Comment (fksyuan) [inline contracts/WorkerMgt.sol]: The event must include worker id.

File: contracts/WorkerMgt.sol [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        uint32[] calldata taskTypes,
+        bytes[] calldata publicKey,
+        bytes calldata quorumNumbers,
+        string calldata socket,
+        IBLSApkRegistry.PubkeyRegistrationParams calldata params,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory operatorSignature
+    ) internal {}
+
+    //generate a random number
+    function getRandomNumber() internal returns (uint256) {
```
Comment (fksyuan) [inline contracts/WorkerMgt.sol]: getRandomNumber function name can add _

**Review by fksyuan — CHANGES_REQUESTED** [review]

File: contracts/PADORegistryCoordinator.sol (hunk 1 above)
Comment (xudean) [reply contracts/PADORegistryCoordinator.sol:78]: fixed

File: contracts/PADORegistryCoordinator.sol (hunk 2 above)
Comment (xudean) [reply contracts/PADORegistryCoordinator.sol]: fixed

File: contracts/WorkerMgt.sol (hunk 3 above)
Comment (xudean) [reply contracts/WorkerMgt.sol]: fixed

File: contracts/WorkerMgt.sol (hunk 4 above)
Comment (xudean) [reply contracts/WorkerMgt.sol]: fixed

**Review by fksyuan — APPROVED** [review]

**Pull request merged**